### ISSUE TITLE
bug: Ionic/React IonMenu & IonMenuButton doesn't work for ios modes/types

### ISSUE THREAD

**Issue by burkmarr** [ionic-team/ionic#19917]
# Bug Report

**Ionic version:**
@Ionic/React 4.11.5

**Current behavior:**
```IonMenu``` can be invoked from ```IonMenuButton``` when type attribute is not set or is set to 'overlay'. When it is set to 'reveal' or 'push' the menu does not appear when ```IonMenuButton``` is used.

Furthermore, setting the Ionic mode configuration to 'ios' also results in a non-functioning ```IonMenuButton``` and ```IonMenu```.

**Expected behavior:**
Setting mode of 'ios' or setting the ```IonMenu``` type attribute to 'reveal' or 'push' may change the behaviour/appearance of the ```IonMenuButton``` and ```IonMenu```, but they should still work.

**Steps to reproduce:**
Demonstration here: https://stackblitz.com/edit/react-zu4a5c

Change commenting on lines 39 to 41 to see the different behaviours:
```js
const type='overlay'
//const type='reveal'
//const type='push'
```
You can also change the mode by uncommenting lines 30 to 32 (and refreshing app).
```js
setupConfig({
  mode: 'ios'
})
```

**Comment by elylucas** [comment]
Hi @Burkmarr,

The contentId prop on your menu is targeting the content in the menu itself, but it is supposed to be pointing to an element that contains the page to move. If you move the id="content" from the IonContent tag to the IonPage tag it will work again.

Did you see this somewhere that we need to update the docs for?

Thanks for the issue. I'm going to close it but feel free to comment back.

**Comment by burkmarr** [comment]
Thanks very much @elylucas - I was able to get it to work with that info. You explained the purpose of  the ```contentID``` attribute more clearly above with:

> it is supposed to be pointing to an element that contains the page to move

than the documentation itself does (https://ionicframework.com/docs/api/menu) with:

> The content's id the menu should use

I my problem was that without the ```contentID``` attribute - which I didn't use because the example  React code doesn't reference it - it didn't work at all and when I realised I needed it and put it in (incorrectly pointing to the menu content) it appeared to work, at least for the 'overlay' type, so I assumed I had it correctly but that there was something wrong with the other types. So some clarification in the documentation might prevent someone else making the same mistake.
